### ISSUE TITLE
Spawning prefab in property setter method causes corrupted scene data on save

### ISSUE THREAD

**Issue by W2Wizard** [FlaxEngine/FlaxEngine#104]
Quite a mouthfull of a title but its exactly that.

**Issue description:**
I was trying to call a method inside the setter of a property in order to spawn a specific amount of prefabs while in editor, the ExecuteInEditMode attribute does sort of the same however it won't execute the OnStart method when a value changes.

It works all quite well at first, when changing the slider the prefabs spawn and unspawn as expected. Saving works as well however once closing the scene and re-opening it results in a error being throw and then crashing the editor.

Afterwards trying to reopen the scene results in the same error being prompted, in the end its not possible to open it anymore.

**Steps to reproduce:**
1. Create a new script and paste this section of code into the class:
```csharp
// Prefab to spawn
public Prefab SpawnPrefab;

private int _num;
private List<Actor> _actors = new List<Actor>();

[Range(0, 100)]
public int PrefabNum 
{
    get => _num;

    set
    {
        _num = value;
        SpawnPrefabs();
    }
}

public void SpawnPrefabs()
{
    // Clean up old actors
    foreach (var actor in _actors)
    {
        Destroy(actor);
    }

    // Spawn in new ones
    for (int i = 0; i < _num; i++)
    {
        var prefab = PrefabManager.SpawnPrefab(SpawnPrefab, Vector3.Zero);
        prefab.SetParent(Parent, true);
        _actors.Add(prefab);
    }
}
```
2. Create a new empty actor and a prefab with something in it like a cube, add your script to the actor, assign the prefab and enter a value. The prefabs should then spawn in and get destroyed when you change the slider.

3. After having created some new actors, save the scene, close it and reopen it.

4. Get Error, click Ok, RIP.

![image](https://user-images.githubusercontent.com/63303990/104076550-a7a07100-5216-11eb-82f4-105e3491d773.png)

After this when you try to reopen the scene it will result in the same error prompt being shown and also results in a crash.

**Video:**

https://user-images.githubusercontent.com/63303990/104077999-2d72eb00-521c-11eb-9de4-4364458f9e84.mp4

**Flax version:**
1.0.6215

**Comment by jb-perrier** [comment]
IMO it's a bad idea to do that, the issue is that the editor is serializing the property value and restore it when it need to, so SpawnPrefabs is called by the editor even in non play mode.

**Comment by mafiesto4** [comment]
Fixed some time ago, probably by: 3239150a610d38bccb9a93f786371ed7227fefda or maybe even before.
Also, by adding `prefab.HideFlags = HideFlags.DontSave;` to spawned prefab it will be runtime-only without saving it to the scene file.
